### PR TITLE
Limit migration banner to shipping supported countries and currencies

### LINKED ISSUE #2816: Hide shipping migration banner for ineligible stores
We're currently display the migration banner to all stores no matter country or currency.

* Related to: https://github.com/woocommerce/woocommerce-shipping/issues/769

### PR DESCRIPTION
**Pull request by kallehauge**
## Description

TL;DR: Hide migration banner for all stores not eligible to purchase shipping labels based on country and currency.

We have 2 different migration nudges in WCS&T:

1. The Feature Announcement when someone opens the "purchase shipping label" modal (introduced in: #2743)
2. The banner that appears of settings pages (introduced in: #2771)

Bullet 1 is only being showed to "shipping eligible stores" since it depends on the shipping label modal.
The issue is the migration banner in bullet 2. We're showing the banner to _all_ origin countries and currencies.

### Related issue(s)

* Resolves #2816
* Related to https://github.com/woocommerce/woocommerce-shipping/issues/769

### Steps to reproduce & screenshots/GIFs

1. Spin up a new site
2. Set the store address to be any of the supported countries (e.g. US)
3. Set the store currency to USD
4. Checkout this branch and activate WCS&T
5. Verify you see the migration banner
6. Change the currency to anything not USD
7. Verify the banner is hidden
8. Change currency back to USD
9. Verify it's showing again
10. Change country to any non-supported country (e.g. Denmark or Canada)
11. Verify the banner is hidden

### Checklist

- [ ] unit tests

- [x] `changelog.txt` entry added
- [x] `readme.txt` entry added

### CODE REVIEW

**Review by kloon — APPROVED** [review]
Changes makes sense, and works as described.
